Introducing `stylecheck` linter to detect duplicate package imports in Go code

nickolaev opened the pull request:
We introduce `stylecheck` linter in `.golangci.yaml`, but enable it only for duplicate imports (ST1019)

Then this PR fixes all the duplicate go imports found by running:
```
make golangci-lint
```

This is a re-spin of the original (abandoned) https://github.com/cilium/cilium/pull/28062 by @Archisman-Mridha

Fixes: #27676

brlbil commented on .github/workflows/lint-go.yaml:
```diff
@@ -37,7 +37,7 @@ jobs:
 
   golangci:
-    name: Lint Source Code
+    name: Lint Source Code using golangci
```
Two nits, it would be a more consistent name if we change this to `Lint Source Code Golangci`

brlbil commented on .github/workflows/lint-go.yaml:
```diff
@@ -57,6 +57,24 @@ jobs:
           args: "--out-${NO_FUTURE}format colored-line-number --verbose --modules-download-mode=vendor"
 
+  staticcheck:
+    name: Lint source code using staticcheck (to detect duplicate package imports)
```
and change the name here to `Lint Source Code staticcheck`
these names appear in dashboards so we do not want to bloat them if possible.

Thanks

nathanjsweet reviewed (approved):
:pray:

pippolo84 reviewed (approved):
Thanks! :100:

kaworu commented on .github/workflows/lint-go.yaml:
staticcheck is part of golangci-lint and [is enabled in our configuration](https://github.com/cilium/cilium/blob/7318ce2d0d89a91227e3f313adebce892f3c388e/.golangci.yaml#L112), do we know why the duplicated imports fixed by this PR were not found by the `golangci` GHA job?

mhofstetter replied:
Staticcheck divides its check into three categories (https://staticcheck.dev/docs/checks)

* staticcheck
* stylecheck
* quickfixes

These are also mapped to separate linters within golangci-lint.

* staticcheck: https://golangci-lint.run/usage/linters/#staticcheck
* stylecheck: https://golangci-lint.run/usage/linters/#stylecheck

Currently, only `staticcheck`s are enabled in Ciliums golangci-lint config.

[duplicated imports](https://staticcheck.dev/docs/checks/#ST1019) is part of the `stylecheck` category - hence it should be enough to enable the "stylecheck" linter.

nickolaev replied:
Good point. Adapted the PR to use golangci-lint instead.

kaworu reviewed (commented):
Hubble patch LGTM, however I have a golangci-lint / staticcheck GHA question (see below).

nickolaev committed:
golangci: add style check for duplicate imports ST1019

This will be invoked as part of the CI workflow linting.

Signed-off-by: Nikolay Nikolaev <nikolay.nikolaev@isovalent.com>

mhofstetter reviewed (approved):
LGTM. thanks for switching to golangci-lint. :tada:

kaworu reviewed (approved):
Thank you!

nickolaev committed:
Removing existing duplicate package imports with stylecheck

After adding stylecheck in the previous commit, we do fix all the issues
it found.

The used command was:
make golangci-lint

Signed-off-by: Archisman Mridha <archismanmridha12345@gmail.com>
Signed-off-by: Nikolay Nikolaev <nicknickolaev@gmail.com>

# Conflicts:
#	pkg/k8s/watchers/watcher.go
#	pkg/service/service.go
